src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,265 @@
+        m_torso_X_imu.translation() = _m_torso_X_imu.block<3,1>(0,3);
+      }
+
+      void TorqueOffsetEstimator::computeOffset(const int& nIterations, const double& _epsilon)
+      {
+        if (sensor_offset_status == PRECOMPUTATION) {
+          SEND_MSG("Starting offset computation with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+          n_iterations = nIterations;
+          sensor_offset_status = INPROGRESS;
+        }
+        else if (sensor_offset_status == INPROGRESS) {
+          SEND_MSG("Collecting input signals. Please keep the graph running", MSG_TYPE_WARNING_STREAM);
+        }
+        else if (sensor_offset_status == COMPUTED) {
+          SEND_MSG("Recomputing offset with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+
+          // resetSavedSignals();
+          encSignals.clear();
+          accSignals.clear();
+          gyrSignals.clear();
+          tauSignals.clear();
+
+          jointTorqueOffsets.setZero();
+
+          n_iterations = nIterations;
+          epsilon = _epsilon;
+          sensor_offset_status = INPROGRESS;
+        }
+        else { //should not happend
+          SEND_MSG("Problem in computeOffset sensor_offset_status", MSG_TYPE_ERROR);
+          assert(false && "Problem in computeOffset sensor_offset_status");
+        }
+        return;
+      }
+      
+      DEFINE_SIGNAL_INNER_FUNCTION(collectSensorData, dummy)
+      {
+        if (sensor_offset_status == PRECOMPUTATION)  {
+          SEND_MSG("Currently Precomputation. Not collecting signals", MSG_TYPE_DEBUG);
+        }
+        else if (sensor_offset_status == INPROGRESS) {
+          // Check the current iteration status
+          int _i = encSignals.size();
+
+          if (_i < n_iterations) {
+            SEND_MSG("Collecting signals for iteration no:" + _i , MSG_TYPE_DEBUG);
+            const dynamicgraph::Vector& _enc = m_base6d_encodersSIN(iter);
+            const dynamicgraph::Vector& _acc = m_accelerometerSIN(iter);
+            const dynamicgraph::Vector& _gyr = m_gyroscopeSIN(iter);
+            const dynamicgraph::Vector& _tau = m_jointTorquesSIN(iter);
+
+            encSignals.push_back(_enc);
+            accSignals.push_back(_acc);
+            gyrSignals.push_back(_gyr);
+            tauSignals.push_back(_tau);
+          }
+          else if (_i == n_iterations){
+            // Do the computations, enough signals collected
+            calculateSensorOffsets();
+            sensor_offset_status = COMPUTED;
+          }
+          else if (_i > n_iterations) {
+            SEND_MSG("This should not happen", MSG_TYPE_ERROR);
+            assert(false);
+          }
+          else {
+            SEND_MSG("This should not happen", MSG_TYPE_ERROR);
+            assert(false);
+          }
+        }
+        else if (sensor_offset_status == COMPUTED) {
+          SEND_MSG("Offset computed", MSG_TYPE_DEBUG);
+        } 
+        else {
+          SEND_MSG("This should not happen", MSG_TYPE_ERROR);
```

Comment:
I find all these "This should not happen" a bit pointless. Is there a specific reason for doing that?